tests/tests_integration/test_api/test_sequences.py

```diff
@@ -70,3 +72,27 @@ def test_get_new(self, new_seq):
         res = COGNITE_CLIENT.sequences.retrieve(id=new_seq.id)
         # assert ["DOUBLE"] == res.column_value_types # soon to change
         assert ["column0"] == res.column_external_ids
+
+
+@pytest.mark.dsl
+class TestSequencesPandas:
+    def test_create_from_dataframe(self):
+        import pandas as pd
+
+        extid = "__test_df{}.{}".format(sys.version_info.major, sys.version_info.minor)
+
+        df = pd.DataFrame([[1, 2.3, "a"], [4, 4.1, "b"]])
+        df.columns = ["intcol", "floatcol", "strcol"]
+        df2 = pd.DataFrame([[1, 2.3], [4, 4.1]])
+
+        seq = COGNITE_CLIENT.sequences.create_from_dataframe(dataframe=df, external_id=extid, force_recreate=True)
+        assert 3 == len(seq.columns)
+        assert "LONG" == seq.columns[0]["valueType"]
+        assert "DOUBLE" == seq.columns[1]["valueType"]
+        assert "STRING" == seq.columns[2]["valueType"]
+
+        with pytest.raises(IncompatibleColumnTypesError):
+            COGNITE_CLIENT.sequences.create_from_dataframe(dataframe=df2, external_id=extid)
+
+        COGNITE_CLIENT.sequences.create_from_dataframe(dataframe=df2, external_id=extid, force_recreate=True)
```

Comment:
This only tests that this doesn't raise an exception. Bare minimum would be to retrieve the dataframe and ensure that it is correct.

We also in essence have multiple tests within the same test here, which is not good.

Reply:
we basically never test retrieve right after insert, due to the delay.
the test sets up a state that's necessary for the other tests which makes it hard to disentangle.